### PR TITLE
Preload for Fonts: UI (1/3)

### LINKED ISSUE #2545: Preload for Fonts: UI (1/3)
Sub-cards for epic #2543.

This card is focused on the UI changes to make to add the option on the settings page.

#### Admin\Settings\Page
- [x] `preload_section()`: update `menu_description` in `add_page_section()`
- [x] `preload_section()`: update `title` in `add_settings_sections()` for `preload_section`
- [x] `preload_section()`: add new  item array  in `add_settings_fields()` for the new option

### PR DESCRIPTION
**Pull request by Screenfeed**
Sub-cards for epic #2543.

This card is focused on the UI changes to make to add the option on the settings page.

#### Admin\Settings\Page
- [x] `preload_section()`: update `menu_description` in `add_page_section()`.
- [x] `preload_section()`: update `title` in `add_settings_sections()` for `preload_section`.
- [x] `preload_section()`: add new  item array  in `add_settings_fields()` for the new option.

#### Q&A
- [x] Q&A

Closes #2545.

### CODE REVIEW

**Comment by hellofromtonya** [comment]
@Screenfeed The base is set to `master`. Did you mean `develop` as the base branch???

**Comment by Screenfeed** [comment]
@hellofromtonya The base branch is supposed to be `feature/preload-for-fonts` but Github doesn’t let me use it because there are still no commits in this PR, so this branch and `feature/preload-for-fonts` are the same. Same for `develop` since `feature/preload-for-fonts` is based on `develop`.

**Comment by Screenfeed** [comment]
Notes:
- The "More Info" link has no valid URL yet.
- The beacon must be filled.
- The 3 "Need Help?" links use the same URL (`$bot_beacon['url']`).

File: inc/classes/admin/settings/class-page.php [hunk 1]
```diff
@@ -1150,6 +1161,16 @@ private function preload_section() {
 					'default'           => [],
 					'sanitize_callback' => 'sanitize_textarea',
 				],
+				'preload_fonts' => [
+					'type'              => 'textarea',
+					'label'             => __( 'Fonts to preload', 'rocket' ),
+					'description'       => __( 'Specify urls of the font files to be preloaded (one per line)', 'rocket' ),
```
Comment (Tabrisrp) [inline inc/classes/admin/settings/class-page.php:1167]: Can you add an `helper` entry to also display what was suggested here: https://github.com/wp-media/wp-rocket/issues/2543#issuecomment-615234025

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/classes/admin/settings/class-page.php (hunk 1 above)
Comment (Screenfeed) [reply inc/classes/admin/settings/class-page.php:1167]: @Tabrisrp Done in https://github.com/wp-media/wp-rocket/pull/2554/commits/cd4ead9016d28ce7e6a7cfc04ca864c212672a3f.

**Review by hellofromtonya — CHANGES_REQUESTED** [review]
The `inc/classes/admin/settings/class-page.php` class is touched in this PR. Please move it to the new architecture, rename, and deprecate the old class.

**Comment by Screenfeed** [comment]
There is one issue here: the namespace is `WP_Rocket\Admin\Settings\Page` (with `Admin` starting with a uppercase `A`) but there is already a folder `/inc/admin` (with a lowercase `a`).
Three ways:
- Change the namespace to `WP_Rocket\admin\Settings\Page` and place the file in `/inc/admin/Settings`. Yuck.
- Rename the folder `admin` into `Admin`. Conflicts incoming, especially on MacOS, where the folder will not be renamed, and probably on our clients sites too.
- Change the namespace to `WP_Rocket\Settings\Page` (or something else) and create a folder `Settings`. I don't really see issues here.

What do you think?

**Comment by Tabrisrp** [comment]
You can use the architecture layed-out on this diagram: https://drive.google.com/file/d/1Yd4Im298-I0s-h-BnqfQRdJfMCkMyMSg/view?usp=sharing

`WP_Rocket\Engine\Settings\Page`

**Comment by Screenfeed** [comment]
Thanks Remy. Done in https://github.com/wp-media/wp-rocket/pull/2554/commits/723c4e236ff85786da2399ba301347980f52ec17.

**Review by remyperona — APPROVED** [review]

**Review by hellofromtonya — APPROVED** [review]

**Comment by Tabrisrp** [comment]
@wp-media/dev-qa ready for QA.

This part is only adding the UI on the settings page, so the option itself won't work.

But you can check that all the required changes were made, and we have the correct copywriting for the option.

**Review by arunbasillal — APPROVED** [review]
- UI looks good 👍 
- Tab description for preload tab is updated 👍 

I notice minor improvements that were added (like sorting this list `otf, ttf, svg, woff, woff2.`). Nice 👍 

Thank you.

**Pull request merged**